journal/src/main/scala/com/evolutiongaming/kafka/journal/Journals.scala

```diff
@@ -268,15 +268,28 @@ object Journals {
               eventRecord    <- read(head, stream)
             } yield eventRecord
 
-            stream.stateful(from) { case (seqNr, a) =>
-              if (seqNr <= a.seqNr) {
-                val seqNr1 = a
-                  .seqNr
-                  .next[Option]
-                (seqNr1, Stream[F].single(a))
-              } else {
-                (seqNr.some, Stream[F].empty)
-              }
+            val empty = Option.empty[EventRecord[A]]
+
+            stream.map(Option(_)).concat(Stream.single(empty)).stateful(empty) {
```

Comment:
this all things should be part of `cassandra` specific layer and not of general kafka-journal